Library/Homebrew/dependency_collector.rb

```diff
@@ -84,11 +85,11 @@ def parse_string_spec(spec, tags)
     if HOMEBREW_TAP_FORMULA_REGEX === spec
       TapDependency.new(spec, tags)
     elsif tags.empty?
-      Dependency.new(spec, tags)
+      PrioritizedTapDependency.new("#{@tap}/#{spec}", tags)
```

Comment:
Instead of string concat, you can just pass tap as another argument. i.e. `PrioritizedTapDependency.new(@tap, spec, tags)`. Of course you need to change the construction method accordingly.